core/src/main/java/io/grpc/internal/PickFirstLeafLoadBalancer.java

```diff
@@ -610,27 +612,26 @@ public PickResult pickSubchannel(PickSubchannelArgs args) {
   }
 
   /**
-   * Index as in 'i', the pointer to an entry. Not a "search index."
+   * This contains both an ordered list of addresses and a pointer(i.e. index) to the current entry
    * All updates should be done in a synchronization context.
    */
   @VisibleForTesting
   static final class Index {
-    private List<EquivalentAddressGroup> addressGroups;
-    private int size;
-    private int groupIndex;
-    private int addressIndex;
+    private List<UnwrappedEag> orderedAddresses = new ArrayList<>();
```

Comment:
Why add the initialization here? Isn't this value never used?

Reply:
Good point.  Removed